src/Sparse/Wiedemann.jl

```diff
@@ -0,0 +1,164 @@
+@doc Markdown.doc"""
+    wiedemann(A::SMat{gfp_elem}, TA::SMat{gfp_elem}) ->Vector{gfp_elem}
+Computes ker($A$).
+"""
+function wiedemann(A::SMat{gfp_elem}, TA::SMat{gfp_elem}) #N::fmpz || N::Int64
+	RR = base_ring(A)
+	N = modulus(RR)
+	T = elem_type(RR)
+	(n,m) = nrows(A),ncols(A)
+	# Prealloc +Randomchoice
+    r = rand(RR, m)
+	c = rand(RR, m)
+    randlin = rand_srow(min(m,10),m,min(10,N),RR)
+	seq = Vector{T}(undef, 2*n)
+	storing_n = Vector{T}(undef,n)
+    storing_m =  Vector{T}(undef,m)
+	z = zero(RR)
+    #Wiedemann sequence
+    # solve A^tAx = A^tAr = y => x -r in kernel(A^tA) to avoid finding zero vec
+	y = Hecke.mul!(storing_m,TA, Hecke.mul!(storing_n,A,r))
+	seq[1] = dot(randlin,c,zero!(z)) #randlin*c 		
+	for i = 2:2*n  #Wiedemann sequence 2m?
+        c =  Hecke.mul!(c,TA, Hecke.mul!(storing_n,A,c))
+		seq[i] = dot(randlin,c) 
+	end
+
+	done,f = Hecke_berlekamp_massey(seq)
```

Comment:
I guess this should be `Hecke.berlekamp_massey`? And since this is Hecke, `berlekamp_massey` should be enough.

Reply:
Sorry. I just saw the function.